dbt/include/spark/macros/materializations/incremental.sql

```diff
@@ -60,21 +60,21 @@
 
 {% endmacro %}
 
-{% macro get_merge_sql(source, target, unique_key) %}
-    merge into {{ target }} as DBT_INTERNAL_DEST
-    using {{ source.include(schema=false) }} as DBT_INTERNAL_SOURCE
-    on DBT_INTERNAL_SOURCE.{{ unique_key }} = DBT_INTERNAL_DEST.{{ unique_key }}
-    when matched then update set *
-    when not matched then insert *
+
+{% macro spark__get_merge_sql(target, source, unique_key, dest_columns, predicates=[]) %}
+  {% set merge_source = source.include(schema=false) %}
+  {% do return(default__get_merge_sql(target, source, unique_key, dest_columns, predicates)) %}
```

Comment:
Just a note for myself: I'm not sure if there's any reason that we should prefer Spark's support for concise syntax
```
    when matched then update set *	
    when not matched then insert *
```
over the more explicit version other databases require, where we name every column for update/insert. They're otherwise identical, and I agree that we should defer to the default implementation where possible.

Reply:
If there's no performance benefit to being explicit, we are going to pay the cost of an extra metadata query (getting the columns). It might be reasonable to override this for spark to go back to `*` for $$$ reasons - I'm pretty ignorant of the spark cost structure, in terms of both money and in performance.

Reply:
You're right—the cost of the metadata query is added cluster time, which equates to $$$ more or less directly. I just did some simple testing and confirmed that using `*` works even when the columns in source + dest are in different orders. Which is surprisingly good, and much better than the issues we ran into with `insert_overwrite` (#60).

While we're here, another matter: `default__get_merge_sql` returned almost all the right syntax, but it wrapped columns in the wrong quoting (`"`) [here](https://github.com/fishtown-analytics/dbt/blob/dev/barbara-gittings/core/dbt/include/global_project/macros/materializations/common/merge.sql#L38). That was surprising to me, since we set spark's quote character to be `` ` `` both for [relations](https://github.com/fishtown-analytics/dbt-spark/blob/master/dbt/adapters/spark/relation.py#L24) and for [columns](https://github.com/fishtown-analytics/dbt-spark/blob/master/dbt/adapters/spark/column.py#L32); I think we're missing the `quote` classmethod in `adapters/spark/impl.py`.

Reply:
re: quoting, we're going to have to centralize the darn quote characters at some point, but you are correct - we need to override the `quote` method.